pkg/ccl/backupccl/backup_planning.go

```diff
@@ -674,7 +674,16 @@ func checkPrivilegesForBackup(
 	}
 	for _, desc := range targetDescs {
 		switch desc := desc.(type) {
-		case catalog.DatabaseDescriptor, catalog.TableDescriptor:
+		case catalog.DatabaseDescriptor:
+			// SELECT is being deprecated as privilege on Databases in 22.1.
+			// In the meanwhile, we still allow backup if the user has SELECT.
+			// TODO(richardjcai): Remove this check for SELECT in 22.1.
+			selectErr := p.CheckPrivilege(ctx, desc, privilege.SELECT)
+			connectErr := p.CheckPrivilege(ctx, desc, privilege.CONNECT)
```

Comment:
why not move one of these -- maybe the SELECT one -- inside the err != nil check for the other to avoid the double check if the first succeeds?